### DISCUSSION TITLE
How to custom global not-found page at Internationalization Routes?

### DISCUSSION THREAD

**Discussion by wildfrontend** [vercel/next.js#50475] [App Router] [answered]
as [example](https://nextjs.org/docs/app/building-your-application/routing/internationalization) if set lang path, the root directory will under `[lang]` folder, but when i set custom `not-found.tsx` file under `[lang]`, it would be not work, unless i  use `notfound()` at page componentt or put at `app/not-found`

but if put at `app/not-found` it would not be get config provider(ex: i18nProvider,Ui Provider...) from `Root Layout`  , so how to do that?

**Comment by devjiwonchoi** [comment]
If you set your `not-found.tsx` file on the root directory of your project, it will be wrapped by your root `layout.tsx`.

So your statement:
> but if put at app/not-found it would not get config provider(ex: i18nProvider, Ui Provider...) from Root Layout

is false.

So your project structure would be as follows:
```
[app]
├── layout.tsx
├── not-found.tsx
└── [lang]
    ├── // ...
```

**Reply by wildfrontend** [reply]
OK.. but , how can do root `not-found.tsx` Internationalization  ?  i want a better way to set root custom `not-found` under `[lang]` folder

**Comment by devjiwonchoi** [comment, marked as answer]
As it is stated [here](https://nextjs.org/docs/app/building-your-application/routing/internationalization#routing-overview),
>The root layout can also be nested in the new folder (e.g. app/[lang]/layout.js).

which also lies with `not-found.tsx`, so as you stated before, 
you can add 'Internationalized' `layout.tsx` and `not-found.tsx` under [lang] directory.

For example,
```
[app]
└── [lang]
    ├── layout.tsx
    ├── not-found.tsx
    ├── // ...
```

Note that you need to use the [`notFound()`](https://nextjs.org/docs/app/api-reference/functions/not-found) function if the `not-found.tsx` is not on the root directory.

Example code:
```
import { notFound } from 'next/navigation';
 
async function fetchUsers(id) {
  const res = await fetch('https://...');
  if (!res.ok) return undefined;
  return res.json();
}
 
export default async function Profile({ params }) {
  const user = await fetchUser(params.id);
 
  if (!user) {
    notFound();
  }
 
  // ...
}
```
ref: https://nextjs.org/docs/app/api-reference/functions/not-found
